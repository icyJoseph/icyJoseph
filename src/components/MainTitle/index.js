import React, { Component } from "react";
import PropTypes from "prop-types";
import { Container, Header } from "semantic-ui-react";
import ReactRevealText from "react-reveal-text";

class MainTitle extends Component {
  state = { reveal: false };

  revealTitle = this.revealTitle.bind(this);

  componentDidMount() {
    const { timeout = 1000 } = this.props;
    this.timer = setTimeout(this.revealTitle, timeout);
  }

  componentWillUnmount() {
    clearTimeout(this.timer);
  }

  revealTitle() {
    return this.setState({ reveal: true });
  }

  render() {
    const {
      title,
      desktop,
      timeout = 500,
      delayMin = 200,
      delayMax = 1200
    } = this.props;
    return (
      <Container style={style.container}>
        <Header as="h1" size="huge" textAlign="center">
          <ReactRevealText
            show={this.state.reveal}
            style={{
              ...style.title,
              letterSpacing: desktop ? "0.7em" : "0.5em",
              color: desktop ? "black" : "FloralWhite"
            }}
            transitionTime={timeout}
            delayMin={delayMin}
            delayMax={delayMax}
          >
            {title}
          </ReactRevealText>
        </Header>
      </Container>
    );
  }
}

MainTitle.propTypes = {
  title: PropTypes.string.isRequired,
  desktop: PropTypes.bool
};

const style = {
  container: {
    display: "flex",
    flex: 1,
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    minHeight: "90px"
  },
  title: {
    fontSize: "35px",
    lineHeight: "36px"
  }
};

export default MainTitle;
