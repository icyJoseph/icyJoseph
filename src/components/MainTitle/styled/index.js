import styled from "styled-components";

export const TitleContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: 90px;
`;
