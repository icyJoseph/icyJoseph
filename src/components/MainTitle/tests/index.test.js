import React from "react";
import { shallow } from "enzyme";
import MainTitle from "../";

describe("MainTitle", () => {
  const props = { title: "Test Title" };
  const spy = jest.spyOn(MainTitle.prototype, "componentDidMount");
  const revealSpy = jest.spyOn(MainTitle.prototype, "revealTitle");
  const wrapper = shallow(<MainTitle {...props} />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
    expect(spy).toHaveBeenCalled();
  });

  it("changes state when revealSpy is called", () => {
    const initialState = wrapper.state(["reveal"]);
    wrapper.instance().revealTitle();
    expect(wrapper.state(["reveal"])).toEqual(!initialState);
  });
});
