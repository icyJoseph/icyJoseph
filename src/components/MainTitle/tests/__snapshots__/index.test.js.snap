// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainTitle renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainTitle
    title="Test Title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Header
        as="h1"
        textAlign="center"
>
        <ReactRevealText
                className=""
                delayMax={1200}
                delayMin={200}
                show={false}
                style={Object {}}
                text=""
                threshold={0.2}
                timingFunction="linear"
                transitionTime={1300}
        >
                Test Title
        </ReactRevealText>
</Header>,
      "style": Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "minHeight": "150px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "h1",
        "children": <ReactRevealText
          className=""
          delayMax={1200}
          delayMin={200}
          show={false}
          style={Object {}}
          text=""
          threshold={0.2}
          timingFunction="linear"
          transitionTime={1300}
>
          Test Title
</ReactRevealText>,
        "textAlign": "center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test Title",
          "className": "",
          "delayMax": 1200,
          "delayMin": 200,
          "show": false,
          "style": Object {},
          "text": "",
          "threshold": 0.2,
          "timingFunction": "linear",
          "transitionTime": 1300,
        },
        "ref": null,
        "rendered": "Test Title",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Header
          as="h1"
          textAlign="center"
>
          <ReactRevealText
                    className=""
                    delayMax={1200}
                    delayMin={200}
                    show={false}
                    style={Object {}}
                    text=""
                    threshold={0.2}
                    timingFunction="linear"
                    transitionTime={1300}
          >
                    Test Title
          </ReactRevealText>
</Header>,
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "minHeight": "150px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": <ReactRevealText
            className=""
            delayMax={1200}
            delayMin={200}
            show={false}
            style={Object {}}
            text=""
            threshold={0.2}
            timingFunction="linear"
            transitionTime={1300}
>
            Test Title
</ReactRevealText>,
          "textAlign": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test Title",
            "className": "",
            "delayMax": 1200,
            "delayMin": 200,
            "show": false,
            "style": Object {},
            "text": "",
            "threshold": 0.2,
            "timingFunction": "linear",
            "transitionTime": 1300,
          },
          "ref": null,
          "rendered": "Test Title",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
